### PR TITLE
apigee x trial provision script

### PR DESCRIPTION
**Pull request by laughingbiscuit**
@yuriylesyuk created a script in `ahr` that I would have found very useful at the weekend, but i didnt not see it.

With some trivial changes it is suitable for `devrel` _without_ a pipeline. Provisioning an X org takes ~ 12-15 minutes and we probably dont want to do this in a nightly pipeline. 

I have assigned Yuriy as codeowner as I have added no original content, only autofixes/patches from `shellcheck`, `markdownlint` and `licensecheck`.

(edit) the reason this adds value over the original docs, is the ingress configuration which isn't clear in the current docs. [ref](https://community.apigee.com/questions/89815/apigee-x-trial-how-to-call-hello-world-proxy.html)

(edit) note that the self signed certs  in line 133 can be expanded in future for a production ready setup, however as a starting point this is a useful reference to publish and iterate on rather than waiting.

- [x] I have run all the tests locally and they all pass.
- [x] I have followed the relevant style guide for my changes.

**CC:** @apigee-devrel-reviewers

### CODE REVIEW

File: tools/apigee-x-trial-provision/apigee-x-trial-provision.sh [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+fi
+
+
+# Step 1: Define functions and environment variables
+function token { echo -n "$(gcloud config config-helper --force-auth-refresh | grep access_token | grep -o -E '[^ ]+$')" ; }
+
+PROJECT_NUMBER=$(gcloud projects describe "$PROJECT" --format="value(projectNumber)")
+export PROJECT_NUMBER
+
+export NETWORK=default
```
Comment (danistrebel) [inline tools/apigee-x-trial-provision/apigee-x-trial-provision.sh:32]: can we make NETWORK and SUBNET overridable as well?

File: tools/apigee-x-trial-provision/README.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Apigee X Trial Provisioning
+
+> Note - this script is not subject to pipelines
+
+## Description
+
+A reference example of [Command Line](https://cloud.google.com/apigee/docs/api-platform/get-started/install-cli)
+Provisioning for Apigee X.
+
```
Comment (danistrebel) [inline tools/apigee-x-trial-provision/README.md:9]: Maybe include list of overridable default values in here?

File: README.md [hunk 3]
```diff
@@ -58,6 +58,8 @@ Apigee products.
   A quickstart setup configuration for Apigee hybrid on GKE
 - [Decrypt Hybrid Assets](tools/decrypt-hybrid-assets) -
   A rescue utility to decrypt proxies and shared flows
+- [ApigeeX Trial Provisioning Reference](tools/apigee-x-trial-provision) -
```
Comment (danistrebel) [inline README.md:61]: Trivial Apigee[SPACE]X

File: tools/apigee-x-trial-provision/apigee-x-trial-provision.sh [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+
+export MIG=apigee-envoy-$REGION
+
+# 
+export RUNTIME_SSL_CERT=~/mig-cert.pem
+export RUNTIME_SSL_KEY=~/mig-key.pem
+export RUNTIME_HOST_ALIAS=$ORG-eval.apigee.net
+
+
+# Step 2: Enable APIs
```
Comment (danistrebel) [inline tools/apigee-x-trial-provision/apigee-x-trial-provision.sh:49]: Should print these step headers to STDOUT? Just from a usability perspective. Emojis optional ;).

File: tools/apigee-x-trial-provision/apigee-x-trial-provision.sh [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+## 4. Create a new eval org using the following command:
+
+gcloud alpha apigee organizations provision \
+  --runtime-location="$REGION" \
+  --analytics-region="$AX_REGION" \
+  --authorized-network=default \
+  --project="$PROJECT"
+
+
+# Step 7: Configure routing, EXTERNAL
```
Comment (danistrebel) [inline tools/apigee-x-trial-provision/apigee-x-trial-provision.sh:73]: Inconsistent step counting. Missing 5&6. I'd scratch the numbers.

Comment (danistrebel) [inline tools/apigee-x-trial-provision/apigee-x-trial-provision.sh:73]: (optional) make EXTERNAL exposure optional

File: tools/apigee-x-trial-provision/apigee-x-trial-provision.sh [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+echo ""
+echo "# To send a test request, execute following commands:"
+echo ""
+echo "export RUNTIME_IP=$RUNTIME_IP"
+
+echo "export RUNTIME_SSL_CERT=~/mig-cert.pem"
+echo "export RUNTIME_SSL_KEY=~/mig-key.pem"
+echo "export RUNTIME_HOST_ALIAS=$RUNTIME_HOST_ALIAS"
+
+echo 'curl --cacert $RUNTIME_SSL_CERT https://$RUNTIME_HOST_ALIAS/hello-world -v --resolve "$RUNTIME_HOST_ALIAS:443:$RUNTIME_IP"'
```
Comment (danistrebel) [inline tools/apigee-x-trial-provision/apigee-x-trial-provision.sh:185]: remind me, do we automatically deploy a hello world proxy to every X eval org?

**Review by danistrebel — CHANGES_REQUESTED** [review]

File: tools/apigee-x-trial-provision/apigee-x-trial-provision.sh (hunk 6 above)
Comment (laughingbiscuit) [reply tools/apigee-x-trial-provision/apigee-x-trial-provision.sh:185]: image
yup!

File: tools/apigee-x-trial-provision/apigee-x-trial-provision.sh (hunk 1 above)
Comment (laughingbiscuit) [reply tools/apigee-x-trial-provision/apigee-x-trial-provision.sh]: fixed

File: tools/apigee-x-trial-provision/README.md (hunk 2 above)
Comment (laughingbiscuit) [reply tools/apigee-x-trial-provision/README.md:9]: fixed and included caveat. this is mainly for reference and will be improved in future (variable validation, proper certs etc.)

File: README.md (hunk 3 above)
Comment (laughingbiscuit) [reply README.md]: done

File: tools/apigee-x-trial-provision/apigee-x-trial-provision.sh (hunk 5 above)
Comment (laughingbiscuit) [reply tools/apigee-x-trial-provision/apigee-x-trial-provision.sh:73]: these numbers currently map to the documentation. i agree that in future we will clean these up, however given how fresh apigee X is, when the docs change I believe these comments will speed up our refactoring

Comment (laughingbiscuit) [reply tools/apigee-x-trial-provision/apigee-x-trial-provision.sh:73]: raising an issue for this enhancement

Comment (laughingbiscuit) [reply tools/apigee-x-trial-provision/apigee-x-trial-provision.sh:73]: https://github.com/apigee/devrel/issues/186

Comment (laughingbiscuit) [reply tools/apigee-x-trial-provision/apigee-x-trial-provision.sh:73]: https://github.com/apigee/devrel/issues/187

File: tools/apigee-x-trial-provision/apigee-x-trial-provision.sh (hunk 4 above)
Comment (laughingbiscuit) [reply tools/apigee-x-trial-provision/apigee-x-trial-provision.sh:49]: added as future enhancement: https://github.com/apigee/devrel/issues/187

for now we use `set -x` for observability.

**Review by danistrebel — APPROVED** [review]
LGTM

**Review by seymen — APPROVED** [review]

**Pull request merged**